tests/kafkatest/tests/core/zookeeper_authorizer_test.py

```diff
@@ -0,0 +1,98 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.cluster.remoteaccount import RemoteCommandError
+from ducktape.mark import matrix
+from ducktape.mark.resource import cluster
+from ducktape.tests.test import Test
+
+from kafkatest.services.kafka import KafkaService, quorum
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.security.kafka_acls import ACLs
+
+class ZooKeeperAuthorizerTest(Test):
```

Comment:
We have a few system tests that exercise acls. Can you please describe why we need a new one? Not saying we don't, but it's good to be clear with regards to the gap we're trying to fill.

Also, we may want to document that we intend to replace the ZK authorizer with a KRaft one. When that happens, I assume we would have a new class for testing the latter and this would be changed to only test zk mode?

Reply:
The only two system tests that leverage an authorizer are `ZooKeeperSecurityUpgradeTest` and `TestSecurityRollingUpgrade`.  The first is specific to ZooKeeper and does not apply to KRaft at all.  The second could be applied to KRaft because it tests changing the inter-broker security protocol across rolls rather than any broker version upgrade, but it has not yet been converted to the KRaft case, plus it does not remove ACLs (which we also want to test here).  Also, the authorizer piece at the end is not really the main part of that test.  It felt appropriate to create this simple, straightforward test for the specific use of the ZooKeeper-based authorizer.

I've added a comment to this test (assuming we keep it) stating that we will test the KRaft replacement authorizer separately.